### PR TITLE
Move to julia 1.11

### PR DESCRIPTION
**Pull request by benegee**
Change to current julia version 1.11.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -205,6 +200,7 @@ jobs:
         run: |
           cd build
           make -j2
+          du -hL ./prefix-pc/lib/libtrixi.so
```
Comment (sloede) [inline .github/workflows/ci.yml:203]: ?

**Comment by benegee** [comment]
CI currently fails because of a memory issue with p4est examples related to the new t8code version in Trixi. @jmark 
I observed this with julia 1.10.6 as well. For unknown reasons it happens with the github runners only when using julia 1.11.

**Comment by benegee** [comment]
While @jmark fights relentlessly with GC, MPI, and t8code, we can get away with finalizing P4est meshes explicitly.
https://github.com/trixi-framework/libtrixi/blob/fb276cc6557260ff5eda7b8557fa07bd18b5d5ff/LibTrixi.jl/src/api_jl.jl#L55

**Review by sloede — APPROVED** [review]

File: .github/workflows/ci.yml
[the remark this answers is not in the document]
Comment (benegee) [reply .github/workflows/ci.yml:1]: Done!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -36,7 +36,7 @@ jobs:
         arch:
           - x64
         julia_version:
-          - '1.10'
+          - '1.11'
```
[the remark this answers is not in the document]
Comment (benegee) [reply .github/workflows/ci.yml:40]: Nice to see both versions running. 1.10 is faster...

[the remark this answers is not in the document]
Comment (benegee) [reply .github/workflows/ci.yml:40]: Should we maybe disable converage testing for 1.10? 
(Shouldn't running both confuse codecov/coveralls)

[the remark this answers is not in the document]
Comment (sloede) [reply .github/workflows/ci.yml:40]: I agree. Coverage is sufficient for the latest verison.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -22,7 +22,7 @@ on:
 
 jobs:
   test:
-    name: ${{ matrix.os }} - ${{ matrix.test_type }} - ${{ github.event_name }}
+    name: ${{ matrix.os }} - ${{ matrix.test_type }} - ${{ matrix.julia_version }} - ${{ github.event_name }}
```
Comment (sloede) [inline .github/workflows/ci.yml:25]:
```suggestion
    name: ${{ matrix.os }} - ${{ matrix.test_type }} - Julia ${{ matrix.julia_version }} - ${{ github.event_name }}
```

**Review by sloede — APPROVED** [review]

**Pull request merged**